File: WebClientIS/WebClientIS/Models/Distrito.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace APIIS.Models
{
    public class Distrito
    {
        public int IdD { get; set; }

        public string Nome { get; set; }

        public Distrito()
        {
            IdD = 0;
            Nome = string.Empty;
        }

        

        public List<Distrito> GetDistritos()
        {
            List<Distrito> lista = new List<Distrito>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Distrito");

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        lista.Add(obj.ToObject<Distrito>());
                    }
                    catch (Exception ex)
                    {
                        return new List<Distrito>();
                    }
                }

            }

            return lista;
        }

        public bool CreateDistrito(Distrito dis)
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, string>
            {
                { "IdD", dis.IdD.ToString() },
                { "Nome", dis.Nome }


            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Distrito");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: MobileClient/sampleproject/lib/Models/Distrito.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';

class Distrito extends Model{

  int idd;
  String nome;

  Distrito({int iddd, String name})
  {
    this.idd = iddd;
    this.nome = name;
  }

   factory Distrito.fromJson(Map<String,dynamic> json){

    return Distrito(
      iddd: json['idD'] as int,
      name: json['nome'] as String
          
    );

  }

   Future<List<Distrito>> getdistritos() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Distrito"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Distrito> listau = new List<Distrito>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Distrito.fromJson(lista[i]));
  

  return listau;
}

Future<int> createdistrito(Distrito dis) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/FatoresDeRisco';
  var body = json.encode(<String, String>{
      'IdD': dis.idd.toString(),
      'Nome':dis.nome

    });


   http.Response response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;

}

}